Tangem/Modules/ManageTokens/ManageTokensViewModel.swift

```diff
@@ -10,4 +10,161 @@ import Foundation
+            .sink { [weak self] string in
+                if !string.isEmpty {
+                    Analytics.log(.tokenSearched)
+                }
+
+                self?.loader.fetch(string)
+            }
+            .store(in: &bag)
+    }
+
+    func setupListDataLoader() -> ListDataLoader {
+        let supportedBlockchains = Set(userWalletRepository.models.map { $0.config.supportedBlockchains }.joined())
+        let loader = ListDataLoader(supportedBlockchains: supportedBlockchains)
+
+        loader.$items
+            .receive(on: DispatchQueue.main)
+            .sink(receiveValue: { [weak self] items in
+                guard let self = self else {
+                    return
+                }
+
+                let alreadyUpdateQuoteCoinIds = tokenViewModels.filter {
+                    $0.priceChangeState != .loading || $0.priceChangeState != .noData
+                }.map { $0.id }
+
+                print(tokenViewModels.map { $0.priceChangeState })
+
+                print(items.count)
+                print(alreadyUpdateQuoteCoinIds.count)
+
+                tokenViewModels = items.compactMap { self.mapToTokenViewModel(coinModel: $0) }
+                updateQuote(by: items.filter { !alreadyUpdateQuoteCoinIds.contains($0.id) }.map { $0.id })
+            })
+            .store(in: &bag)
+
+        return loader
+    }
+
+    func sendAnalyticsOnChangeTokenState(tokenIsSelected: Bool, tokenItem: TokenItem) {
+        Analytics.log(event: .tokenSwitcherChanged, params: [
+            .state: Analytics.ParameterValue.toggleState(for: tokenIsSelected).rawValue,
+            .token: tokenItem.currencySymbol,
+        ])
+    }
+
+    // MARK: - Private Implementation
+
+    private func displayAlert(title: String, message: String) {
+        let okButton = Alert.Button.default(Text(Localization.commonOk))
+
+        alert = AlertBinder(alert: Alert(
+            title: Text(title),
+            message: Text(message),
+            dismissButton: okButton
+        ))
+    }
+
+    private func actionType(for coinModel: CoinModel) -> ManageTokensItemViewModel.Action {
+        let userWalletModels = userWalletRepository.models
+
+        let isAlreadyExistToken = userWalletModels.contains(where: { userWalletModel in
+            coinModel.items.contains(where: { tokenItem in
+                userWalletModel.userTokensManager.contains(tokenItem, derivationPath: nil)
+
+            })
+        })
+
+        return isAlreadyExistToken ? .edit : .add
+    }
+
+    private func mapToTokenViewModel(coinModel: CoinModel) -> ManageTokensItemViewModel {
+        ManageTokensItemViewModel(
+            coinModel: coinModel,
+            action: actionType(for: coinModel),
+            didTapAction: handle(action:with:)
+        )
+    }
+
+    private func updateQuote(by coinIds: [String]) {
+        tokenQuotesRepository
+            .loadQuotes(coinIds: coinIds)
+            .sink()
+            .store(in: &bag)
+    }
```

Comment:
Давай лучше отдельную подписку сделаем что бы она отменялась если новая пришла
```suggestion
        self.loadQuotesSubscribtion = tokenQuotesRepository
            .loadQuotes(coinIds: coinIds)
            .sink()
    }
```